tests/unit/retrievers/test_base.py

```diff
@@ -37,12 +41,62 @@ def test_retriever_version_support(
     expected_exception: Union[type[ValueError], None],
 ) -> None:
     class MockRetriever(Retriever):
-        def _get_search_results(self, *args: Any, **kwargs: Any) -> None:  # type: ignore
-            pass
+        def _get_search_results(self, *args: Any, **kwargs: Any) -> RawSearchResult:
+            return RawSearchResult(records=[])
 
     driver.execute_query.return_value = [[{"versions": db_version}], None, None]
     if expected_exception:
         with pytest.raises(expected_exception):
             MockRetriever(driver=driver)
     else:
         MockRetriever(driver=driver)
+
+
+@patch("neo4j_genai.retrievers.base.Retriever._verify_version")
+def test_retriever_search_docstring_copied(
+    _verify_version_mock: MagicMock,
+    driver: MagicMock,
+) -> None:
+    class MockRetriever(Retriever):
+        def _get_search_results(self, query: str, top_k: int = 10) -> RawSearchResult:
```

Comment:
Should `_get_search_results` be a public method since it's optionally customisable by users?

Reply:
I think it used to be public and we updated it because it is not callable from outside the class :) I do not have a strong opinion on this one TBH, could be public for the reason you describe.

Reply:
Now that I think about it, having it private makes it confusing for the user as to whether it should be implemented. It should probably be public like in the first place, wdyt?

Reply:
Yes makes sense to me, private also tends to be synonym of "do not touch". I'll update it to public and add a comment so that we try not to change it again in the future :)

Reply:
And actually, other methods in this class with the same goal, like `get_result_formatter` were not private, so it's better in terms of consistency.

Done in https://github.com/neo4j/neo4j-genai-python/pull/68/commits/f3a589b193b2c720ff9853344c08a6878cdc247e